### PR TITLE
CSV download of country tools

### PR DESCRIPTION
**Pull request by reinhardt**
syslabcom/scrum#838

### CODE REVIEW

File: src/euphorie/content/browser/country.py [hunk 1]
```diff
@@ -101,3 +104,38 @@ class Tools(CountryView):
...
+        fieldnames = [
+            "Sector",
+            "Tool Title",
+            "Version Title",
+            "URL",
+            "published",
+            "obsolete",
+        ]
+        buffer = StringIO()
+        writer = csv.DictWriter(buffer, fieldnames=fieldnames, dialect="bilbomatica")
```
Comment (thet) [inline src/euphorie/content/browser/country.py:118]: why the dialect `bilbomatica`?‌ the default is `excel` which would be what users most often expect and what the ticket description asks for.
If there is a reason, please leave a comment in code.

File: src/euphorie/content/browser/country.py [hunk 2]
```diff
@@ -101,3 +104,38 @@ class Tools(CountryView):
...
+                            "published": "published" if survey["published"] else "",
+                            "obsolete": "obsolete" if group["obsolete"] else "",
+                        }
+                    )
+        csv_data = buffer.getvalue()
+        buffer.close()
+        response = self.request.RESPONSE
+        response.setHeader(
+            "Content-Disposition",
+            f"attachment; filename=oira_tools_{self.context.id}.csv",
```
Comment (thet) [inline src/euphorie/content/browser/country.py:138]: just a matter of taste, should we include the date isostring also in the filename? I personally would do so.

File: src/euphorie/content/browser/templates/country_tools.pt [hunk 3]
```diff
@@ -13,6 +13,7 @@
     <metal:title fill-slot="title">${view/title}</metal:title>
     <metal:content fill-slot="content">
       <h2 i18n:translate="header_sector_survey_list">OiRA Tools</h2>
+      <a class="button" href="${context/absolute_url}/@@country-tools.csv" i18n:translate="header_sector_survey_download">Download spreadsheet</a>
```
Comment (thet) [inline src/euphorie/content/browser/templates/country_tools.pt:16]: :+1: :)

**Review by thet — CHANGES_REQUESTED** [review]
All fine, except for clarification for the csv dialect and a suggestion which you may just ignore.

File: src/euphorie/content/browser/country.py (hunk 1 above)
Comment (reinhardt) [reply src/euphorie/content/browser/country.py:118]: I went for consistency with [existing code](https://github.com/euphorie/Euphorie/blob/country-tools-csv/src/euphorie/content/utils.py#L27), but thinking about it again, you're probably right and the default is better if nothing specific was requested.

**Comment by reinhardt** [comment]
Pushed new version with default dialect and date in file name.

**Review by thet — APPROVED** [review]
:+1:

**Pull request merged**